ha/core/event_analyzer/parser/parser.py

```diff
@@ -188,9 +198,18 @@ def parse_event(self, msg: str) -> HealthEvent:
             event_type = cluster_resource_alert[EventAttr.EVENT_PAYLOAD.value][HealthAttr.RESOURCE_STATUS.value]
             specific_info = cluster_resource_alert[EventAttr.EVENT_PAYLOAD.value][HealthAttr.SPECIFIC_INFO.value]
             if resource_type == CLUSTER_ELEMENTS.NODE.value:
+                node_type = None
+                if node_id in self.data_node_ids:
+                    node_type = NODE_TYPE.DATA.value
+                elif node_id in self.control_node_ids:
+                    node_type = NODE_TYPE.CONTROL.value
+                elif node_id in self.server_node_ids:
+                    node_type = NODE_TYPE.SERVER.value
```

Comment:
Can we have these in a separate function? like,
`def get_functional_type(resource_type)`

Reply:
creating function def get_functional_type(node_id)